### ISSUE TITLE
Expose what extras are available for install

### ISSUE THREAD

**Issue by dstufft** [pypa/warehouse#1855]
This comes from https://github.com/pypa/pip/issues/2718, it will likely require some ecosystem level changes to make happen but we should expose it when we can.

**Comment by brainwane** [comment]
In our [meeting today](https://wiki.python.org/psf/PackagingWG/2018-02-12-Warehouse) we decided this nice-to-have feature belongs in a future milestone.

**Comment by miketheman** [comment]
Somewhat related: #11380

**Comment by webknjaz** [comment]
I was just looking at https://pypi.org/p/id and in the sidebar, it does have a

> **Meta**
> **License:** Apache Software License
> **Requires:** Python >=3.9
> **Provides-Extra:** `dev` , `lint` , `test`

So I was thinking that this issue should be reframed to "show install commands for extras". I think the UI could be a drop-down next to the normal `pip install <pkg> <copy-btn>` that would show additional variants like `pip install '<pkg> [extra1]' <copy-btn>` and `pip install '<pkg> [extra1, extra2, extra3, all-the-rest]' <copy-btn>`.
